### PR TITLE
Update DropboxWatchService.java

### PR DESCRIPTION
**Pull request by rtdog**
added features to support interruption for more consistency with the FileSystem one.

### CODE REVIEW

**Comment by rtdog** [comment]
@FJBDev, these are the changes I'm recommending for better support in the EasyImportManager code on MT.  I don't think there was anything wrong with the existing code, but I added some features to more explicitly support interrupts and to make the behavior more consistent with the FileSystemWatchService which makes the logic in EasyImportManager simpler.

**Comment by FJBDev** [comment]
@rtdog Thanks for your changes, I am fine with them. Did you actually test those locally ? or you only coded and compiled ? 
If/When you have tested them, let me know and I will approve your PR, build an updated jar and put that in MT 20.8

**Comment by rtdog** [comment]
@FJBDev , yes I tested them pretty thoroughly but only through MT, I did not look into test framework on the java7-fs-dropbox codestream, so if there are unit and/or regression tests in there they have not been executed.

I actually wasn't sure the best way to manage both of these projects in Eclipse at the same time and would welcome any recommendation on how to do it.  I pulled the java7-fs-dropbox project from GitHub, added it as another project in Eclipse, then removed the existing .jar file from net.tourbook.cloud and created a "jardesc" file in Eclipse so I could recompile the JAR and stuff the output into the net.tourbook.cloud directory by double-clicking that "jardesc" file.  I feel like there should be a way to improve that build process but couldn't figure it out.  Any suggestions there would be appreciated and let me know if there are unit/regression tests that need executed.  Thanks!

**Comment by FJBDev** [comment]
@rtdog 

> yes I tested them pretty thoroughly but only through MT, I did not look into test framework on the java7-fs-dropbox codestream, so if there are unit and/or regression tests in there they have not been executed.

Ok, that sounds good. As long as you tested your code changes in MT, that you ensured that there are no regressions and that the expected behavior happens, I am good with that.
There are tests in the java7-fs-dropbox it self that can get run when you build that jar.
I build it as below (as you can see it's simple :-) ) :

`gradlew build`

I will run them for safety's sake but they are not thorough so I prefer to test in MT.

I will build the new jar asap and do a PR into MT so it can get into 20.8

Thanks!

PS: 

>  I feel like there should be a way to improve that build process but couldn't figure it out

I agree but so far, I thought that the best would be to leave this project on its own and update the jar in MT only when needed.
It's not a terribly complicated process so I'm ok sticking with it for now.

File: src/main/java/com/github/fge/fs/dropbox/driver/DropboxWatchService.java [hunk 1]
```diff
@@ -192,24 +195,26 @@ public WatchKey poll(final long timeout, final TimeUnit unit) throws Interrupted
...
       while (_hasChanges.get() == false && _continuePolling) {
-         try {
-
+         try {        	 
+             if (_httpClient == null) {            	 
+            	 throw new ClosedWatchServiceException();
+             }      
             final String cursor = getLatestCursor("");
 
-            //  final ListFolderLongpollResult listFolderLongpollResult = DropboxClient.getDefault().files().listFolderLongpoll(cursor);
```
Comment (FJBDev) [inline src/main/java/com/github/fge/fs/dropbox/driver/DropboxWatchService.java:207]: @rtdog  I will keep those commented lines as it was on purpose.
If Dropbox fulfills [this request in their SDK](https://github.com/FJBDev/java7-fs-dropbox/blob/d5ce6f59e7f4f1e31e0dd8694778995bed3d9f98/src/main/java/com/github/fge/fs/dropbox/driver/DropboxWatchService.java#L135), then I can uncomment this code

**Pull request merged**